src/Compilers/CSharp/Portable/Symbols/ExtraAnnotations.cs

```diff
@@ -1,181 +0,0 @@
-{
-    // https://github.com/dotnet/roslyn/issues/29821 external annotations should be removed or fully designed/productized
-    //  If we choose to stick with an ad-hoc key (rather than annotations as source or as PE/ref assembly),
-    //  we should consider the assembly qualified name format used in metadata (with backticks and such).
-    internal static class ExtraAnnotations
-    {
-        // APIs that are useful to annotate:
-        //   1) don't accept null input
-        //   2) return a reference type
-        // All types in a member which can be annotated should be annotated. Value types and void can be skipped (with a `default`)
-        private static readonly ImmutableDictionary<string, ImmutableArray<ImmutableArray<byte>>> Annotations =
-            new Dictionary<string, ImmutableArray<ImmutableArray<byte>>>
-            {
-                { "System.Boolean System.Boolean.Parse(System.String)", Array(default, Nullable(false)) },
-                { "System.Void System.Buffer.BlockCopy(System.Array, System.Int32, System.Array, System.Int32, System.Int32)", Array(default, Nullable(false), default, Nullable(false), default, default) },
-                { "System.Int32 System.Buffer.ByteLength(System.Array)", Array(default, Nullable(false)) },
-                { "System.Byte System.Buffer.GetByte(System.Array, System.Int32)", Array(default, Nullable(false), default) },
-                { "System.Void System.Buffer.SetByte(System.Array, System.Int32, System.Byte)", Array(default, Nullable(false), default, default) },
-                { "System.Byte System.Byte.Parse(System.String)", Array(default, Nullable(false)) },
-                { "System.Byte System.Byte.Parse(System.String, System.IFormatProvider)", Array(default, Nullable(false), default) },
-                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles)", Array(default, Nullable(false), Nullable(false)) },
-                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles, System.IFormatProvider)", Array(default, Nullable(false), Nullable(false), default) },
-                { "System.String System.String.Concat(System.String, System.String)", Array(Nullable(false), Nullable(true), Nullable(true)) },
-            }.ToImmutableDictionary();
-
-        private static readonly ImmutableDictionary<string, ImmutableArray<FlowAnalysisAnnotations>> Attributes =
-            new Dictionary<string, ImmutableArray<FlowAnalysisAnnotations>>
-            {
-                { "System.Boolean System.String.IsNullOrEmpty(System.String)", Array(default, NotNullWhenFalse) },
-                { "System.Boolean System.String.IsNullOrWhiteSpace(System.String)", Array(default, NotNullWhenFalse) },
-                { "System.Boolean System.String.Contains(System.String)", Array(default, EnsuresNotNull) },
-                { "System.Void System.Diagnostics.Debug.Assert(System.Boolean)", Array(default, AssertsTrue) },
```

Comment:
>System.Void System.Diagnostics.Debug.Assert(System.Boolean)", Array(default, AssertsTrue) [](start = 19, length = 89)

Should the compiler treat `Debug.Assert(bool condition, ...)` as `Debug.Assert([AssertsTrue] bool condition, ...)`, even when unannotated?

Reply:
No. That is the whole point of this change.

---
In reply to: [274941794](https://github.com/dotnet/roslyn/pull/34916#discussion_r274941794) [](ancestors = 274941794)